frontend/frontend-filters.php

```diff
@@ -237,17 +237,20 @@ public function sort_widgets_sidebars( $sidebars_widgets, $wp_registered_widgets
 	 *
 	 * @since 3.1
 	 *
-	 * @param array $sidebars_widgets        An associative array of sidebars and their widgets
-	 * @param  array $wp_registered_widgets  Array of all registered widgets.
+	 * @param  array  $sidebars_widgets      An associative array of sidebars and their widgets
+	 * @param  array  $wp_registered_widgets Array of all registered widgets.
+	 * @param  object $instance              Instance of class that handles widgets blocks filters.
 	 * @return array|null                    An associative array of sidebars and their widgets or nothing.
 	 */
-	public function init_sidebars_widgets( $sidebars_widgets, $wp_registered_widgets ) {
+	public function init_sidebars_widgets( $sidebars_widgets, $wp_registered_widgets, $instance = null ) {
 		if ( empty( $wp_registered_widgets ) ) {
 			return $sidebars_widgets;
 		}
 
-		$this->cache_key         = md5( maybe_serialize( $sidebars_widgets ) );
-		$_sidebars_widgets = $this->cache->get( "sidebars_widgets_{$this->cache_key}" );
+		$instance = $instance ? $instance : $this;
```

Comment:
This is needed only because of `PLL_Widgets_Editor_Frontend_Fitlers`.

Since we have decided to use one cache per class, maybe this separation into function should be thought out again?